src/signal.rs

```diff
@@ -1903,4 +1928,170 @@ mod menu_item {
+                let f: Box<Box<Fn(&Self, &TreeModel, &TreeIter, bool, bool) + 'static>> =
+                    Box::new(Box::new(apply_attributes_func));
+                connect(self.to_glib_none().0, "apply-attributes",
+                    transmute(apply_trampoline::<Self> as usize), Box::into_raw(f) as *mut _)
+            }
+        }
+
+        fn connect_focus_changed<F>(&self, focus_changed_func: F) -> u64
+        where F: Fn(&Self, &CellRenderer, &str) + 'static {
+            unsafe {
+                let f: Box<Box<Fn(&Self, &CellRenderer, &str) + 'static>> =
+                    Box::new(Box::new(focus_changed_func));
+                connect(self.to_glib_none().0, "focus-changed",
+                    transmute(focus_trampoline::<Self> as usize), Box::into_raw(f) as *mut _)
+            }
+        }
+
+        fn connect_remove_editable<F>(&self, remove_editable_func: F) -> u64
+        where F: Fn(&Self, &CellRenderer, &CellEditable) + 'static {
+            unsafe {
+                let f: Box<Box<Fn(&Self, &CellRenderer, &CellEditable) + 'static>> =
+                    Box::new(Box::new(remove_editable_func));
+                connect(self.to_glib_none().0, "remove-editable",
+                    transmute(remove_trampoline::<Self> as usize), Box::into_raw(f) as *mut _)
+            }
+        }
+    }
+
+    unsafe extern "C" fn add_trampoline<T>(this: *mut GtkCellArea,
+                                           renderer: *mut GtkCellRenderer,
+                                           editable: *mut GtkCellEditable,
+                                           cell_area: *mut GdkRectangle,
+                                           path: *const c_char,
+                                           f: &Box<Fn(&T, &CellRenderer, &CellEditable, &Rectangle, &str) + 'static>)
+    where T: IsA<CellArea> {
+        callback_guard!();
+        let buf = CStr::from_ptr(path).to_bytes();
+        let string = str::from_utf8(buf).unwrap();
+        f(&CellArea::from_glib_none(this).downcast_unchecked(),
+          &CellRenderer::from_glib_none(renderer),
+          &CellEditable::from_glib_none(editable),
+          &Rectangle { x: (*cell_area).x, y: (*cell_area).y, width: (*cell_area).width, height: (*cell_area).height },
+          string);
+    }
+
+    unsafe extern "C" fn apply_trampoline<T>(this: *mut GtkCellArea,
+                                             model: *mut GtkTreeModel,
+                                             iter: *mut GtkTreeIter,
+                                             is_expander: gboolean,
+                                             is_expanded: gboolean,
+                                             f: &Box<Fn(&T, &TreeModel, &TreeIter, bool, bool) + 'static>)
+    where T: IsA<CellArea> {
+        callback_guard!();
+        f(&CellArea::from_glib_none(this).downcast_unchecked(),
+          &TreeModel::from_glib_none(model).downcast_unchecked(),
+          &TreeIter::from_glib_none(iter),
```

Comment:
From the existing trampolines in this module it seems that `TreeIter` should be converted with `from_glib_borrow`.